src/test/java/org/springframework/integration/kafka/listener/AbstractBrokerTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.springframework.integration.kafka.listener;
+
+import static scala.collection.JavaConversions.asScalaBuffer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import com.gs.collections.api.RichIterable;
+import com.gs.collections.api.block.function.Function2;
+import com.gs.collections.api.multimap.Multimap;
+import com.gs.collections.api.multimap.MutableMultimap;
+import com.gs.collections.api.tuple.Pair;
+import com.gs.collections.impl.factory.Multimaps;
+import com.gs.collections.impl.tuple.Tuples;
+import kafka.admin.AdminUtils;
+import kafka.producer.KeyedMessage;
+import kafka.producer.Producer;
+import kafka.producer.ProducerConfig;
+import kafka.serializer.StringEncoder;
+import kafka.utils.TestUtils;
+import org.junit.After;
+import scala.collection.JavaConversions;
+import scala.collection.Map;
+import scala.collection.immutable.List$;
+import scala.collection.immutable.Map$;
+import scala.collection.immutable.Seq;
+
+import org.springframework.integration.kafka.core.Configuration;
+import org.springframework.integration.kafka.core.ConnectionFactory;
+import org.springframework.integration.kafka.core.DefaultConnectionFactory;
+import org.springframework.integration.kafka.core.BrokerAddressListConfiguration;
+
+/**
+ * @author Marius Bogoevici
+ */
+public abstract class AbstractBrokerTest {
+
+	public static final String TEST_TOPIC = "test-topic";
+
+	public abstract KafkaEmbeddedBrokerRule getKafkaRule();
+
+	@After
+	public void cleanUpTopic() {
+		AdminUtils.deleteTopic(getKafkaRule().getZookeeperClient(), TEST_TOPIC);
+		TestUtils.waitUntilMetadataIsPropagated(asScalaBuffer(getKafkaRule().getKafkaServers()), TEST_TOPIC, 0, 5000L);
+	}
+
+	public void createTopic(String topicName, int partitionCount, int brokers, int replication) {
```

Comment:
`@SuppressWarnings("unchecked")`
The `toKafkaPartitionMap` returns raw `scala.collection.Map` without generics.
To be honest I haven't been able to make it generic :smile: 
So, `@SuppressWarnings` is a good compromise